pkg/instancetype/find/spec_test.go

```diff
@@ -0,0 +1,350 @@
+				// Set the following values to be compatible with objects converted from v1alpha1
+				Model:                 pointer.P(""),
+				DedicatedCPUPlacement: pointer.P(false),
+				IsolateEmulatorThread: pointer.P(false),
+			}
+
+			specData, err := json.Marshal(clusterInstancetype.Spec)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Do not set APIVersion as part of VirtualMachineInstancetypeSpecRevision in order to trigger bug #9261
+			specRevision := v1alpha1.VirtualMachineInstancetypeSpecRevision{
+				Spec: specData,
+			}
+			specRevisionData, err := json.Marshal(specRevision)
+			Expect(err).ToNot(HaveOccurred())
+
+			controllerRevision := &appsv1.ControllerRevision{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "crName",
+					Namespace:       vm.Namespace,
+					OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(vm, v1.VirtualMachineGroupVersionKind)},
+				},
+				Data: runtime.RawExtension{
+					Raw: specRevisionData,
+				},
+			}
+
+			_, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Create(
+				context.Background(), controllerRevision, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm.Spec.Instancetype.RevisionName = controllerRevision.Name
+
+			foundInstancetypeSpec, err := finder.Find(vm)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundInstancetypeSpec).To(HaveValue(Equal(clusterInstancetype.Spec)))
+		})
+	})
+
+	Context("Using namespaced Instancetype", func() {
+		var fakeInstancetype *v1beta1.VirtualMachineInstancetype
+
+		BeforeEach(func() {
+			fakeInstancetype = &v1beta1.VirtualMachineInstancetype{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-instancetype",
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: v1beta1.VirtualMachineInstancetypeSpec{
+					CPU: v1beta1.CPUInstancetype{
+						Guest: uint32(2),
+					},
+					Memory: v1beta1.MemoryInstancetype{
+						Guest: resource.MustParse("128Mi"),
+					},
+				},
+			}
+
+			_, err := virtClient.VirtualMachineInstancetype(metav1.NamespaceDefault).Create(
+				context.Background(), fakeInstancetype, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			err = instancetypeInformerStore.Add(fakeInstancetype)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm = libvmi.NewVirtualMachine(libvmi.New(
+				libvmi.WithNamespace(metav1.NamespaceDefault)),
+				libvmi.WithInstancetype(fakeInstancetype.Name),
+			)
+		})
+
+		It("find returns expected instancetype", func() {
+			instancetypeSpec, err := finder.Find(vm)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*instancetypeSpec).To(HaveValue(Equal(fakeInstancetype.Spec)))
+		})
+
+		It("uses client when instancetype not found within informer", func() {
+			err := instancetypeInformerStore.Delete(fakeInstancetype)
+			Expect(err).ToNot(HaveOccurred())
+			instancetypeSpec, err := finder.Find(vm)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*instancetypeSpec).To(HaveValue(Equal(fakeInstancetype.Spec)))
```

Comment:
```suggestion
			Expect(instancetypeSpec).To(HaveValue(Equal(fakeInstancetype.Spec)))
```

With `HaveValue` you should no longer need to deref the ptr.